parser.py

```diff
@@ -9,23 +9,52 @@
 import sys
 from os import listdir
 from os.path import isfile, join
-nltk.download('punkt')
+from collections import defaultdict
+import json
 
 def GetLanguageId(language):
-    language_id_list = logic.SQLQuery("select Lang_ID from lang_ref where Lang_Desc = '"+ language +"'")
+    language_id_list = logic.SQLQuery("select Lang_ID from lang_ref where Lang_Desc = '" + language + "'")
 
-    # Condition for language not present in database
     if len(language_id_list) == 0:
         return -1
     else:
         return language_id_list[0]
 
+def StoreIndexing(dictionary):
+    f = open("indexing.txt", "w")
+    f.write(dictionary)
+    f.close()
+
+def CreateIndexing(sentence, line_id, dictionary):
+    word_list = sentence.split()
+    # number of parts formed for each tagged word in the corpus(word/pos)
+    tagged_word_parts = 2   
+    
+    for word in word_list:
+        # skip untagged words
+        if(len(word.split('/')) != tagged_word_parts):
+            continue
+
+        subtag = (word.split('/')[1]).upper()
+        # ignore -TL suffix in subtags since they are just to indicate that the word occurs in title
+        subtag = subtag.split('-TL')[0]
+
+        if(subtag in logic.english_pos_mapping):
+            generalized_pos = logic.english_pos_mapping[subtag]
+        else:
+            generalized_pos = subtag
+        
+        word = (word.split('/')[0] + "/" + generalized_pos).lower()
```

Comment:
Can you explain, why are we lower casing the words?

Reply:
Since a user may enter the word in random case...so we convert to lower at both ends - searching for the user input word as well as while storing it in the index file.

Reply:
Would be the same for other corpora as well?